src/app/courses/courses.component.ts

```diff
@@ -2,22 +2,32 @@ import { Component, OnInit, AfterViewInit, ViewChild } from '@angular/core';
 import { CouchService } from '../shared/couchdb.service';
 import { DialogsDeleteComponent } from '../shared/dialogs/dialogs-delete.component';
 import { MatTableDataSource, MatSort, MatPaginator, MatFormField, MatFormFieldControl, MatDialog } from '@angular/material';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { SelectionModel } from '@angular/cdk/collections';
+import { Location } from '@angular/common';
 
 @Component({
-  templateUrl: './courses.component.html'
+  templateUrl: './courses.component.html',
+  styles: [ `
+    .space-container {
+      margin: 64px 30px;
+    }
+` ]
 })
-export class CoursesComponent implements OnInit, AfterViewInit {
 
+export class CoursesComponent implements OnInit, AfterViewInit {
+  selection = new SelectionModel(true, []);
   courses = new MatTableDataSource();
   @ViewChild(MatSort) sort: MatSort;
   @ViewChild(MatPaginator) paginator: MatPaginator;
-  displayedColumns = [ 'title', 'action' ];
+  displayedColumns = [ 'select', 'title', 'action' ];
   message = '';
   deleteDialog: any;
 
   constructor(
     private couchService: CouchService,
-    private dialog: MatDialog
+    private dialog: MatDialog,
+    private location: Location
   ) { }
 
   ngOnInit() {
```

Comment:
If the function is not using, then we can remove it . So, Could you please check  it once ?

Reply:
@nirojdyola  there is actually a code inside if you look after the whole file